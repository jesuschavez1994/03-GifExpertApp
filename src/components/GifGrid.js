import React from 'react';
import PropTypes from 'prop-types';
import GifGridItem from './GifGridItem';

import { useFecthGifs } from '../hooks/useFecthGifs';

const GifGrid = ( {category} ) => {

    const { data: imgs, loading } = useFecthGifs( category );

    return (
        <>

        <h3> { category } </h3>

        { loading ? 'Loading...' : 'Data Cargada'}

        <div  className="card-grid">
           {
                imgs.map( img => (
                    <GifGridItem 
                        key={ img.id }
                        { ...img }
                    />
                ))
            }
        </div>   
        


        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid;

// apy_key = Yv6sw29HkE0IhCmxaDYRImsW7dxGU4L9