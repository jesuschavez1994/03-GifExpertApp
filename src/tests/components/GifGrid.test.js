import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import GifGrid from '../../components/GifGrid';
import { useFecthGifs } from '../../hooks/useFecthGifs';
jest.mock('../../hooks/useFecthGifs')


describe('Pruebas en <GifGrid />', () => {

    const category = 'Goku'

    test('Debe de mostrarse correctamente ', () => {

        useFecthGifs.mockReturnValue({
            data: [],
            loading: true
        });

        let wrapper = shallow(<GifGrid category = { category }/>);

        expect( wrapper).toMatchSnapshot();
    });

    test('Deben de mostrar items cuando se cargen Imagenes useFecthGifs', () => {

        const gifs = [
            {
                id: 'ABC',
                url: 'https://localhost/123',
                title: 'title'
            },
            {
                id: 'DEF',
                url: 'https://localhost/123',
                title: 'title'
            }
        ]

        useFecthGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        let wrapper = shallow(<GifGrid category = { category }/>);
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe( false );
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length )
        
    })
    
})
