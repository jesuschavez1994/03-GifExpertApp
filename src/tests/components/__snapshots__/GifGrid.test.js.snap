// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe de mostrarse correctamente  1`] = `
<Fragment>
  <h3>
     
    Goku
     
  </h3>
  Loading...
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid /> Deben de mostrar items cuando se cargen Imagenes useFecthGifs 1`] = `
<Fragment>
  <h3>
     
    Goku
     
  </h3>
  Data Cargada
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="title"
      url="https://localhost/123"
    />
    <GifGridItem
      id="DEF"
      key="DEF"
      title="title"
      url="https://localhost/123"
    />
  </div>
</Fragment>
`;
